src/Cache.php

```diff
@@ -45,18 +47,28 @@ class Cache implements MiddlewareInterface
      */
     protected $mustRevalidate;
 
+    /**
+     * @var StreamFactoryInterface
+     */
+    protected $streamFactory;
+
     /**
      * Create new HTTP cache
      *
      * @param string $type           The cache type: "public" or "private"
      * @param int    $maxAge         The maximum age of client-side cache
      * @param bool   $mustRevalidate must-revalidate
      */
-    public function __construct(string $type = 'private', int $maxAge = 86400, bool $mustRevalidate = false)
-    {
+    public function __construct(
+        StreamFactoryInterface $streamFactory,
+        string $type = 'private',
+        int $maxAge = 86400,
+        bool $mustRevalidate = false
+    ) {
         $this->type = $type;
         $this->maxAge = $maxAge;
         $this->mustRevalidate = $mustRevalidate;
+        $this->streamFactory = $streamFactory;
```

Comment:
Maybe for now it's better to add `slim/psr7` and call `StreamFactory` this would make it easier for anyone already using this package?

Reply:
I'd rather require `psr/http-factory` (and likely `psr/http-factory-implementatio`) than `slim/psr7` as it keeps interoperable with other psr7 implementation packages

Reply:
Only the user should decide which StreamFactoryInterface implementation should be used in their application.

Reply:
@flavioheleno @odan thank you for taken the time to review the PR I’ll update when I get some time.